dali/pipeline/pipeline.cc

```diff
@@ -873,10 +895,24 @@ int Pipeline::output_ndim(int id) const {
   return output_descs_[id].ndim;
 }
 
+
+static bool is_input_operator(const OpNode &op_node) {
+  return (op_node.op_type == OpType::CPU &&
+          dynamic_cast<InputOperator<CPUBackend> *>(op_node.op.get())) ||
+         (op_node.op_type == OpType::MIXED &&
+          dynamic_cast<InputOperator<MixedBackend> *>(op_node.op.get())) ||
+         (op_node.op_type == OpType::GPU &&
+          dynamic_cast<InputOperator<GPUBackend> *>(op_node.op.get()));
+}
+
+
 int Pipeline::num_inputs() const {
-  return ext_input_names_.size();
+  DALI_ENFORCE(built_, "\"Build()\" must be called prior to calling \"num_inputs()\".");
+  auto &op_nodes = graph_.GetOpNodes();
+  return static_cast<int>(std::count_if(op_nodes.begin(), op_nodes.end(), is_input_operator));
```

Comment:
can we calculate this once and keep the result as a member? It could be calculated on pipeline build or similar.

Reply:
Also could you use:
`input_operators_names_.size()`?

Reply:
Yeah, seems like a good idea